### ISSUE TITLE
"Filter unknown extensions" should be related to "Show Advance Settings" 

### ISSUE THREAD

**Issue by sergiobenrocha2** [libretro/RetroArch#4089]
When "Show Advance Settings" OFF, "Filter unknown extensions" should be hide

**Comment by fr500** [comment]
I disagree, a noob user may have some... .fig SNES roms and would be unable to load them.
People have old romsets lingering

**Comment by Kivutar** [comment]
Why don't we add .fig to the list of supported extensions?

**Comment by fr500** [comment]
it was just an example, it could happen for more cases than we can ever handle

**Comment by Alcaro** [comment]
I have never ever seen a FIG ROM. We're not adding that unless you can demonstrate they're commonly used.

And that doesn't solve the underlying problem. What if someone has a .ufo romset instead? Or another of [the 23 extensions snes9x has supported](https://github.com/snes9xgit/snes9x/commit/3e84a2eaf4616e1917cfd53140af2b6c5198007c)?

Or an ancient non-SNES romset? Or multiple romsets, both of which use .usa names?

As for the original issue (whether removing odd extensions is advanced), I'd say yes.

**Comment by bparker06** [comment]
I have an ancient romset from the 90s that uses .FIG for SNES.

**Comment by Kivutar** [comment]
> <Twinaphex> we can hide it behind Advanced Settings
> <Twinaphex> Advanced Settings is ironically on by default anyway
> <Twinaphex> so it’d make no difference
> <Twinaphex> I remember .fig roms for SNES yeah

**Comment by sergiobenrocha2** [comment]
The menu at least should start at the first rom position in the list. This is really unfriendly:

Parent Directory                  <-- menu start here
Filter unknown extensions
foo.rom                               <-- it should start here
foo2.rom
foo3.rom

**Comment by Kivutar** [comment]
it's difficult to do that, and it's a completely different issue that what we're discussing here.

**Comment by jedinjapan** [comment]
Realistically speaking, adding a few custom file extensions to support rare filesets isn't an unreasonable thing to ask any user who wants to hide this menu option, and considers themselves capable enough to do so safely.

Hooking it to advanced settings isn't actually a bad idea in that respect. Wouldn't impact the default install status of Retroarch, and could let users who finish setting up and add their custom settings to .cfg profiles safely get rid of it.
